src/mini-batch.jl

```diff
@@ -0,0 +1,73 @@
+"""
+    MiniBatch(b::Int)
+
+"""
+struct MiniBatch <: AbstractKMeansAlg
+    b::Int  # batch size
+end
+
+
+MiniBatch() = MiniBatch(100)
+
+function km(algo::MiniBatch, X, k;
+                weights = nothing, metric = Euclidean(), n_threads = Threads.nthreads(),
+                k_init = "k-means++", init = nothing, max_iters = 300,
+                tol = 0, verbose = false, rng = Random.GLOBAL_RNG)
+
+    # Step 1. Select sample from X as specified by batch_size and weights
+    nrow, ncol = size(X)  # n_features, m_examples
+    T = eltype(X)
+    rand_idx = isnothing(weights) ? rand(rng, 1:ncol, algo.b) : wsample(rng, 1:ncol, weights, algo.b)
+    initial_batch_sample = X[:, rand_idx]
+
+    # Step 2. Initiate cluster centers from the initial sample
+    centroids = isnothing(init) ? smart_init(initial_batch_sample, k, n_threads, weights, rng, init=k_init).centroids : deepcopy(init)
```

Comment:
I do not think, that it make much sense to `smart_init` from the sample. You should use original data for initial centers, otherwise, you'll be too far away from the destination.